docs/write_tests.rst

```diff
@@ -0,0 +1,131 @@
+^^^^^^^^^^^^^^^
+
+ * Add a folder *test_yourconnectorname* in parsons/test for your connector
+ * Add a file *test_yourconnectorname.py* to the *test_yourconnectorname* folder
+ * Add one `“Happy Path” <https://en.wikipedia.org/wiki/Happy_path>`_ test per public method of your connector
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Tests for HTTP API Connectors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For classes built using the ``APIConnector`` class, we can leverage the ``requests-mock`` library, since the ``APIConnector`` uses the ``requests`` library.
+
+The ``requests-mock`` library allows us to simulate the API we are calling into by pretending to be an HTTP server. We can create canned responses, and introspect on the endpoints that our Connector called to ensure that the method we are testing is acting the way we expect.
+
+For an example of testing a class built on the ``APIConnector``, we can look at the ``MailChimp`` connector.
+
+In the code below, we have our ``TestMailchimp`` that serves as our test case. We have one test method to test calls to our ``get_campaigns`` method in our ``Mailchimp`` Connector class.
+
+We decorate ``test_get_campaigns`` with the `requests_mock.Mocker`` class, which passes in our requests mocker as an argument ``m``. We can then use this mock to configure our mock server.
+
+To configure our mock API, we can make use of the ``get``, ``post``, ``patch``, ``put``, and ``delete`` methods on the mock object ``m``. These methods mirror the calls we expect our Connector to make to the API. The methods on the mock object correspond to the HTTP method that we are expecting our Connector class to call with, and the first argument to the mock method is the URL that we expect our Connector to call. We can then provide a ``json`` argument to the mock method to provide a canned response that the mock HTTP server should return.
+
+Below, we will use the ``get`` method to tell the mock HTTP server that we expect a call to the ``campaigns`` endpoint of our API, and that the mock HTTP server should return our ``test_campaignss` canned response. The Mailchimp tests store the canned responses in a separate Python module, which can help keep the test code separate from the test data, which helps makes tests more readable.
+
+.. code-block:: python
+
+    class TestMailchimp(unittest.TestCase):
+
+        @requests_mock.Mocker()
+        def test_get_campaigns(self, m):
+
+            mc = MailChimp(api_key='12345')
+
+            # Test that campaigns are returned correctly.
+            m.get(self.mc.uri + 'campaigns', json=expected_json.test_campaigns)
+            tbl = self.mc.get_campaigns()
+
+            self.assertEqual(tbl.num_rows, 2)
+
+
+After wiring up the mock HTTP API, we are ready to call the ``get_campaigns`` method on our ``Mailchimp`` connector. Since we are using the ``requests_mock.Mocker`` class, our Connector will not actually hit the Mailchimp API; our call will be intercepted and passed to our mock HTTP server.
```

Comment:
It might be a bit confusing to say "our mock HTTP server" as it isn't really a mock http server, but a canned response. Perhaps something like
```suggestion
After wiring up the mock HTTP API, we are ready to call the ``get_campaigns`` method on our ``Mailchimp`` connector. Since we are using the ``requests_mock.Mocker`` class, our Connector will not actually hit the Mailchimp API; our call will be intercepted and the configured canned response will be returned.
```